src/pie_modules/taskmodules/re_text_classification_with_indices.py

```diff
@@ -1336,7 +1388,21 @@ def create_annotations_from_output(
                     f"creating a new annotation from a candidate_annotation of another type than BinaryRelation is "
                     f"not yet supported. candidate_annotation has the type: {type(candidate_annotation)}"
                 )
-            if not (self.add_candidate_relations and label == self.none_label):
+            # Do not create annotation if
+            # - `add_candidate_relations` is True and relation label is a `none_label`
+            # - `argument_and_relation_type_whitelist` is used and a binary relation is not on a whitelist
+            if not (self.add_candidate_relations and label == self.none_label) and not (
+                isinstance(new_annotation, BinaryRelation)
+                and self.argument_and_relation_type_whitelist is not None
+                and not (
+                    label in self.argument_and_relation_type_whitelist
+                    and (
+                        getattr(new_annotation.head, "label"),
+                        getattr(new_annotation.tail, "label"),
+                    )
+                    in self.argument_and_relation_type_whitelist[label]
+                )
+            ):
```

Comment:
This is super hard to read. Can you move the negations inside the brackets? Also, please use `get_relation_argument_spans_and_roles`.

EDIT: I think it is also fine to have nested `if`s if this is better to read.